### ISSUE TITLE
SS4 PHP7.3 version 3.1 not reporting

### ISSUE THREAD

**Issue by Firesphere** [phptek/silverstripe-sentry#39]
It seems the 3.x branch, with SS4 and PHP7.3 is not reporting anything. 😢

**Comment by phptek** [comment]
Use the 3.0.1 release

**Comment by phptek** [comment]
Be sure to check the README too, a few things have changed, if all you've been using is 3.0.x-dev

**Comment by Firesphere** [comment]
Yep, I've done all the updates and changes, (including using 3.0.1), but it's not doing anything, even if I purposefully throw an error.

To clear any issues, I _can_ connect to Sentry just fine from the prod environment.

Here's my YML:
```
_config# cat sentry.yml 
---
Name: my-project-config-sentry
After:
  - 'sentry-config'
---
PhpTek\Sentry\Adaptor\SentryAdaptor:
  opts:
    dsn: https://mykey:andsentrykey@sentry.io/1722555
```

**Comment by phptek** [comment]
Weird.

Here's mine, and I can confirm it reports A-OK to sentry.io

```
Name: my-project-sentry
After:
  - 'sentry-config'
---

PhpTek\Sentry\Adaptor\SentryAdaptor:
  opts:
    dsn: 'https://0cd144eb2c8246e4a30986bc97488489@sentry.io/1770597'
```

The only difference I see, is that my DSN is quoted. FYI, As of 3.0.1, you can also declare `SENTRY_DSN` in `.env`.

**Comment by phptek** [comment]
Your DSN format looks different to mine though...

Ah....and mine is on php 7.1 though...one sec

**Comment by phptek** [comment]
Works in 7.3 too.

**Comment by Firesphere** [comment]
I are confused now....

I have no clue what's going wrong :/

**Comment by phptek** [comment]
Can you debug the write() method in SentryHandler. If $record is populated with data, in capture message or captureException(), then that's the jump off point to Monolog, Sentry SDK etc.

You can also ensure that your DSN is being propagated correctly by debugging SentryAdaptor::getOpts()...(I think it's SentryAdaptor off the top if my head)

**Comment by Firesphere** [comment]
I got it to work with the `.env` instead. I still have no clue why it wasn't working with the YML though.

**Comment by phptek** [comment]
That is weird, but I will say this after playing around with the config quite a lot yesterday. flush=all doesn't _always_ purge YML config changes. To guarantee you're operating with the YML you think you're operating with, is to `rm -rf /tmp/silverstripe-cache`.

Another "cool" (TM) thing I spotted yesterday was that while playing with YML config, no matter what I did (_including_ purging the cache as above), `Injector` insisted that a non-existent YML config had been set. I struggled with that for an hour and then thought "Is it in memory?". Sure enough, restarting the webserver was what was needed.
